internal/util/util.go

```diff
@@ -133,3 +134,30 @@ func StructToPrettyString(st interface{}) string {
 	s, _ := json.MarshalIndent(st, "", "\t")
 	return string(s)
 }
+
+func ExtractField(data any, path []string) (any, error) {
```

Comment:
If you have time, could add some unit test for this function? 
But only you have time, I know contributing this was not a priority so feel free to leave it as is.